subsys/usb/usb_c/usbc_pe_snk_states.c

```diff
@@ -0,0 +1,839 @@
+		if (atomic_test_bit(&pe->flags, PE_FLAGS_PROTOCOL_ERROR)) {
+			atomic_clear_bit(&pe->flags, PE_FLAGS_PROTOCOL_ERROR);
+		} else {
+			policy_notify(dev, PORT_PARTNER_NOT_RESPONSIVE);
+		}
+		pe_set_state(dev, PE_SNK_HARD_RESET);
+	}
+}
+
+/**
+ * @brief PE_Send_Soft_Reset Exit State
+ */
+void pe_send_soft_reset_exit(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+
+	/* Stop Sender Response Timer */
+	usbc_timer_stop(&pe->pd_t_sender_response);
+}
+
+/**
+ * @brief PE_SNK_Soft_Reset Entry State
+ */
+void pe_soft_reset_entry(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	LOG_INF("PE_SNK_Soft_Reset");
+
+	/* Reset the Protocol Layer */
+	prl_reset(dev);
+	atomic_set_bit(&pe->flags, PE_FLAGS_SEND_SOFT_RESET);
+}
+
+/**
+ * @brief PE_SNK_Soft_Reset Run State
+ */
+void  pe_soft_reset_run(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	if (prl_is_running(dev) == false) {
+		return;
+	}
+
+	if (atomic_test_and_clear_bit(&pe->flags, PE_FLAGS_SEND_SOFT_RESET)) {
+		/* Send Accept message */
+		send_ctrl_msg(dev, PD_PACKET_SOP, PD_CTRL_ACCEPT);
+		return;
+	}
+
+	if (atomic_test_and_clear_bit(&pe->flags, PE_FLAGS_TX_COMPLETE)) {
+		pe_set_state(dev, PE_SNK_WAIT_FOR_CAPABILITIES);
+	} else if (atomic_test_and_clear_bit(&pe->flags, PE_FLAGS_PROTOCOL_ERROR)) {
+		pe_set_state(dev, PE_SNK_HARD_RESET);
+	}
+}
+
+/**
+ * @brief PE_Not_Supported Entry State
+ */
+void pe_send_not_supported_entry(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	LOG_INF("PE_Not_Supported");
+
+	/* Request the Protocol Layer to send a Not_Supported or Reject Message. */
+	if (prl_get_rev(dev, PD_PACKET_SOP) > PD_REV20) {
+		send_ctrl_msg(dev, PD_PACKET_SOP, PD_CTRL_NOT_SUPPORTED);
+	} else {
+		send_ctrl_msg(dev, PD_PACKET_SOP, PD_CTRL_REJECT);
+	}
+}
+
+/**
+ * @brief PE_Not_Supported Run State
+ */
+void pe_send_not_supported_run(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	if (atomic_test_bit(&pe->flags, PE_FLAGS_TX_COMPLETE) ||
+			atomic_test_bit(&pe->flags, PE_FLAGS_MSG_DISCARDED)) {
+		atomic_clear_bit(&pe->flags, PE_FLAGS_TX_COMPLETE);
+		atomic_clear_bit(&pe->flags, PE_FLAGS_MSG_DISCARDED);
+		pe_set_state(dev, PE_SNK_READY);
+	}
+}
+
+/**
+ * @brief PE_Chunk_Received Entry State
+ */
+void pe_chunk_received_entry(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+
+	LOG_INF("PE_SNK_Chunk_Received");
+
+	usbc_timer_start(&pe->pd_t_chunking_not_supported);
+}
+
+/**
+ * @brief PE_Chunk_Received Run State
+ */
+void pe_chunk_received_run(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	if (usbc_timer_expired(&pe->pd_t_chunking_not_supported)) {
+		pe_set_state(dev, PE_SEND_NOT_SUPPORTED);
+	}
+}
```

Comment:
I'm not super familiar with zephyr's timers, but does this one need stopped on exit?

Reply:
I added the exit state that stops the timer